client/src/components/Tool/ToolCard.vue

```diff
@@ -1,295 +1,166 @@
-        },
+    disabled: {
+        type: Boolean,
+        default: false,
     },
-};
+});
+
+const emit = defineEmits(["onChangeVersion"]);
+
+function onChangeVersion(v) {
+    emit("onChangeVersion", v);
+}
+
+const errorText = ref(null);
+
+watch(
+    () => props.id,
+    () => {
+        errorText.value = null;
+    }
+);
+
+function onSetError(e) {
+    errorText.value = e;
+}
+
+const { currentUser: user } = useCurrentUser();
+const hasUser = computed(() => !user.value.isAnonymous);
+
+const versions = computed(() => props.options.versions);
+const showVersions = computed(() => props.options.versions?.length > 1);
 </script>
-<style scoped>
+
+<template>
+    <div class="position-relative">
+        <div class="underlay sticky-top" />
+        <div class="tool-header sticky-top bg-secondary px-2 py-1 rounded">
+            <div class="d-flex justify-content-between">
+                <div class="py-1 d-flex flex-wrap flex-gapx-1">
+                    <span>
+                        <icon icon="wrench" class="fa-fw" />
+                        <h1 itemprop="name">{{ title }}</h1>
+                    </span>
+                    <span itemprop="description">{{ description }}</span>
+                    <span>(Galaxy Version {{ version }})</span>
+                </div>
+                <div class="d-flex flex-nowrap align-items-start">
+                    <b-button-group class="tool-card-buttons">
+                        <ToolFavoriteButton v-if="hasUser" :id="props.id" @onSetError="onSetError" />
+                        <ToolVersionsButton
+                            v-if="showVersions"
+                            :version="props.version"
+                            :versions="versions"
+                            @onChangeVersion="onChangeVersion" />
+                        <ToolOptionsButton :id="props.id" :sharableUrl="props.options.sharable_url" />
+                    </b-button-group>
+                    <slot name="header-buttons" />
+                </div>
+            </div>
+        </div>
+
+        <FormMessage variant="danger" :message="errorText" :persistent="true" />
+        <FormMessage :variant="props.messageVariant" :message="props.messageText" />
+        <slot name="body" />
+        <slot name="buttons" />
+        <div v-if="props.disabled" class="portlet-backdrop" />
+
+        <div>
+            <ToolCardSection title="Help">
+                <ToolHelp :content="props.options.help" />
+            </ToolCardSection>
+
+            <ToolFooter
+                :id="props.id"
+                :has-citations="props.options.hasCitations"
+                :xrefs="props.options.xrefs"
+                :license="props.options.license"
+                :creators="props.options.creators"
+                :requirements="props.options.requirements" />
+        </div>
+    </div>
+</template>
+
+<style lang="scss" scoped>
+@import "scss/theme/blue.scss";
+
+.tool-header {
```

Comment:
Worried about too many of these custom styles here but the rest of the PR looks great to me.